src/WebhookChannel.php

```diff
@@ -45,5 +47,7 @@ public function send($notifiable, Notification $notification)
         if ($response->getStatusCode() >= 300 || $response->getStatusCode() < 200) {
             throw CouldNotSendNotification::serviceRespondedWithAnError($response);
```

Comment:
I still like the idea of setting the response on the exception. Are you able to do that?

Reply:
Ya, but not sure I can do it and maintain that `serviceRespondedWithAnError` method. Seems like I'd need to construct it differently and add a `getResponse` method. But maybe I'm wrong?

Reply:
Something like this

```php
use GuzzleHttp\Psr7\Response;

class CouldNotSendNotification extends \Exception
{
    /** @var Response */
    private $response;

    public function __construct(Response $response, string $message, int $code = null)
    {
        $this->response = $response;
        $this->message = $message;
        $this->code = $code ?? $response->getStatusCode();
    }


    public static function serviceRespondedWithAnError(Response $response)
    {
        return new self(
            $response,
            sprintf('Webhook responded with an error: `%s`', $response->getBody()->getContents())
        );
    }
}
```

Reply:
Done.